src/main/java/edu/hm/hafner/analysis/parser/SimulinkCheckParser.java

```diff
@@ -0,0 +1,116 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.File;
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * A parser for Simulink Check tool by Mathworks results.
+ * Used for .html report files.
+ *
+ * @author Eva Habeeb
+ */
+public class SimulinkCheckParser extends IssuePropertiesParser {
+
+    private static final long serialVersionUID = -8099258658775128275L;
+
+    @Override
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        String name = readerFactory.getFileName();
+        File file = new File(name);
```

Comment:
Use the reader factory to read the file. The filename might not be set correctly.

Reply:
There is no function in redearFactrory that reads the file and returns object with type 'File' which is required for the Jsoup.parse function in line 33.

Reply:
Why can't you use one of the other methods that actually work on a `Reader` or `InputStream`?

Reply:
Which methods do you mean, the methods in the ReaderFactory class ?

Reply:
Jsoup has also methods that accept a `Reader`, and the factory can create a `Reader` as well.

Reply:
I tried to use parse method from Jsoup that accepts 'InputStream' : parse(InputStream in, String charsetName, String baseUri), but the 'shouldHandleEmptyFile' test fails. 
Jsoup doesnt have a parse method that accepts 'Reader' type so I had to convert from 'Reader' to 'InputStream'. (code snapshot attached) 
![image](https://user-images.githubusercontent.com/117662592/210756685-34c1f9da-bac8-4739-8fde-4abcdc263dfd.png)

Reply:
This looks good! When the test fails then you need to fix the code: empty files should either throw a `ParsingException` or an empty list of issues.

Is the encoding always UTF-8? If not: the reader factory already has the correct encoding stored.

Reply:
BTW: does JSOUP load included elements of the HTML document from the internet? How does JSOUP handle JS in the HTML document? Do we need to add XSS protection as we are doing for XML parsers?

Reply:
Added a ParsingException to handle empty files. As for the encoding, Ive check for all the sample files I have and it is always UTF-8, but just incase I changed it to use getCharset() from ReaderFactory.

Reply:
![image](https://user-images.githubusercontent.com/117662592/211567467-ccac8d83-d3c7-4333-b414-d1eab28a37e7.png)

According to the documentation of JSOUP, to prevent XSS attacks I can use their Cleaner class which is:
 "a safelist based HTML cleaner. Use to ensure that end-user provided HTML contains only the elements and attributes that you are expecting; no junk, and no cross-site scripting attacks!
Safe-lists define what HTML (elements and attributes) to allow through the cleaner. Everything else is removed"

Reply:
This looks like a solution to get the HTML clean. What I mean: does JSOUP load content from the internet when placed in the HTML? If yet it might make sense to use the cleaner before the parsing. But if JSOUP does not load anything else (only the provided document) then we are on the safe side.

Reply:
I am not able to find any information on whether JSOUP loads the internet content inside the HTML while using the parse function. The documentation states that the parse function 'reads input stream and parses it into Document'.

![image](https://user-images.githubusercontent.com/117662592/211792019-c1c8a217-6797-4780-949b-bd7701e6837d.png)

I believe it wont be possible to use cleaner before parsing because the cleaner expects an 'org.jsoup.nodes.Document'.

Reply:
Ok, thanks for checking!